scene/gui/control.cpp

```diff
@@ -1732,31 +1732,26 @@ void Control::_size_changed() {
 		new_size_cache.height = minimum_size.height;
 	}
 
-	bool pos_changed = new_pos_cache != data.pos_cache;
-	bool size_changed = new_size_cache != data.size_cache;
+	bool pos_changed = !new_pos_cache.is_equal_approx(data.pos_cache);
+	bool size_changed = !new_size_cache.is_equal_approx(data.size_cache);
 
-	data.pos_cache = new_pos_cache;
-	data.size_cache = new_size_cache;
+	if (pos_changed) {
+		data.pos_cache = new_pos_cache;
+		_notify_transform();
+		item_rect_changed(false);
```

Comment:
What's the impact of calling these twice in the case where `pos_changed && size_changed && is_inside_tree()`? That seems to be the only logic change after the refactoring.

Reply:
That would emit the signal two times, which shouldn't be. I did some changes so that the code is as similar as possible to how it was before.